bson/primitive/primitive.go

```diff
@@ -151,9 +161,9 @@ func (tp Timestamp) IsZero() bool {
 	return tp.T == 0 && tp.I == 0
 }
 
-// CompareTimestamp returns an integer comparing two Timestamps, where T is compared first, followed by I.
-// Returns 0 if tp = tp2, 1 if tp > tp2, -1 if tp < tp2.
-func CompareTimestamp(tp, tp2 Timestamp) int {
+// Compare compares the time instant tp with tp2. If tp is before tp2, it returns -1; if tp is after
+// tp2, it returns +1; if they're the same, it returns 0.
+func (tp Timestamp) Compare(tp2 Timestamp) int {
```

Comment:
We could clean up the logic in compare to use the new "Before" and "After" methods:

```go
func (tp Timestamp) Compare(tp2 Timestamp) int {
	switch {
	case tp.Equal(tp2):
		return 0
	case tp.Before(tp2):
		return -1
	default:
		return +1
	}
}
```

Reply:
Good idea. Updated.